app/services/scene-collections/scene-collections.ts

```diff
@@ -0,0 +1,865 @@
+    const collectionName = this.suggestName(name);
+    await this.loadOverlay(pathName, collectionName);
+  }
+
+  /**
+   * Install a new overlay from a URL
+   * @param filePath the location of the overlay file
+   * @param name the name of the overlay
+   */
+  async loadOverlay(filePath: string, name: string) {
+    this.startLoadingOperation();
+    await this.deloadCurrentApplicationState();
+    await this.overlaysPersistenceService.loadOverlay(filePath);
+    this.setupDefaultAudio();
+
+    const id: string = uuid();
+    await this.insertCollection(id, name);
+    await this.setActiveCollection(id);
+    this.finishLoadingOperation();
+  }
+
+  /**
+   * Based on the provided name, suggest a new name that does
+   * not conflict with any current name.
+   *
+   * Name conflicts are actually ok in this system, but can
+   * be a little confusing for the user, so we soft-enforce
+   * it in the UI layer.
+   * @param name the base name
+   */
+  suggestName(name: string) {
+    return namingHelpers.suggestName(name, (name: string) => {
+      return !!this.collections.find(collection => {
+        return collection.name === name;
+      });
+    });
+  }
+
+  /**
+   * Show the window to name a new scene collection
+   * @param options options
+   */
+  showNameConfig(
+    options: { sceneCollectionToDuplicate?: string; rename?: boolean } = {}
+  ) {
+    this.windowsService.showWindow({
+      componentName: 'NameSceneCollection',
+      queryParams: {
+        sceneCollectionToDuplicate: options.sceneCollectionToDuplicate,
+        rename: options.rename ? 'true' : ''
+      },
+      size: {
+        width: 400,
+        height: 250
+      }
+    });
+  }
+
+  /**
+   * Used by StreamDeck
+   */
+  fetchSceneCollectionsSchema(): Promise<ISceneCollectionSchema[]> {
+    const promises: Promise<ISceneCollectionSchema>[] = [];
+
+    this.collections.forEach(collection => {
+      promises.push(
+        new Promise<ISceneCollectionSchema>(resolve => {
+          const file = path.join(this.collectionsDirectory, `${collection.id}.json`);
+          this.readCollectionFile(file).then(data => {
+            const root = parse(data, NODE_TYPES);
+            const collectionSchema: ISceneCollectionSchema = {
+              id: collection.id,
+              name: collection.name,
+
+              scenes: root.data.scenes.data.items.map(
+                (sceneData: ISceneSchema) => {
+                  return {
+                    id: sceneData.id,
+                    name: sceneData.name,
+                    sceneItems: sceneData.sceneItems.data.items.map(
+                      (sceneItemData: ISceneItemInfo) => {
+                        return {
+                          sceneItemId: sceneItemData.id,
+                          sourceId: sceneItemData.sourceId
+                        };
+                      }
+                    )
+                  };
+                }
+              ),
```

Comment:
omg, is it autoformatting? Looks like the Lisp hell

Reply:
Haha yeah I guess so.  This function does have a lot of indentation, which the auto formatter seems to strange things with.